docs/template-development.md

```diff
@@ -0,0 +1,159 @@
+
+The following examples show some advanced configurations that we can use in our `package.json` file:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react",
+    "supportedProtocols": "mqtt"
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+Based on these custom configurations we can use this template to generate an output. The above `package.json` file has a newly added configuration called `supportedProtocols` which is set to `mqtt`. This configuration displays all the protocols that this template supports. We can have multiple supported protocols in our template. 
+
+For example, if you want to generate an output using above template, you need to have asyncapi document that has servers with `mqtt` in order to generate your desired output. If your asyncapi document has server connections with `kafka`, the generation process will be halted since the only supported protocol mentioned is `mqtt`. 
+
+Additionally, we can also have a configuration called `parameters` which is an object with all the parameters that can be passed when generating the template. When using the command line, it's done by indicating `--param name=value` or `-p name=value`:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react",
+    "supportedProtocols": "mqtt",
+    "parameters": {
+        "version": {
+          "description": "Override the version of your application provided under `info.version` location in the specification file.",
+          "required": false
+        }
+    }
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+The default version of your application is always fetched from your asyncapi document. The above configuration will help the template user to override the existing version with a new version on the command line. For example, `-p version=2.0.0`.
+
+The changes done in the template will be as follows:
+
+Original:
+
+```js
+<Text>App name is: **{ asyncapi.info().title() }**</Text>
+```
+
+Newer:
+
+```js
+<Text>App name is: **{ asyncapi.info().title() }**</Text>
+<Text>Version is: **{params.version || asyncapi.info.version()}**</Text>
+```
+
+## Hooks
+
+[Hooks](hooks.md) enable templates to perform multiple tasks. You can add Hooks to your template as fractions of code. In the template, you must store it in the `hooks` directory under the template directory. You can also store it in other modules and external libraries, or even configure it inside the template. The generation process can perform multiple actions. _(Example: A hook that generates a pdf after the generation process is complete.)_
+
+**Templates** can perform multiple actions _before_ or _after_ the generation process with the help of **hooks**.
+
+For example, In AsyncAPI html-template, hooks can help you change the specification version with the new `version` that you can pass before the generation process even begins:
```

Comment:
`html-template`

don't mention html-template

rather focus on referring only to example that is already done in previous sections